### PR TITLE
Add Android utilities for app file restoration

### PR DESCRIPTION
**Pull request by Leland-Takamine**

### CODE REVIEW

File: maestro-client/src/main/java/maestro/android/AndroidBuildToolsDirectory.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package maestro.android
+
+import java.io.File
+import java.util.regex.Pattern
+
+internal object AndroidBuildToolsDirectory {
+
+    fun findBuildToolsDir(androidHome: File): File {
+        val buildToolsParent = File(androidHome, "build-tools")
```
Comment (dmitry-zaitsev) [inline maestro-client/src/main/java/maestro/android/AndroidBuildToolsDirectory.kt:9]: This makes Maestro dependent on Android SDK, contrary to what we've been saying. Let's make sure this is not a required step for running a simple flow on an already debuggable app

File: maestro-client/src/main/java/maestro/android/ApkDebuggable.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                    override fun child(ns: String?, name: String?): NodeVisitor {
+                        if (name != "application") return super.child(ns, name)
+
+                        return object : NodeVisitor(super.child(ns, name)) {
+
+                            override fun end() {
+                                super.attr(
+                                    "http://schemas.android.com/apk/res/android",
+                                    "debuggable",
+                                    16842767,
```
Comment (dmitry-zaitsev) [inline maestro-client/src/main/java/maestro/android/ApkDebuggable.kt:66]: Some comment on what those number mean would be nice

File: maestro-client/src/main/java/maestro/utils/TemporaryDirectory.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package maestro.utils
+
+import java.nio.file.Files
+import java.nio.file.Path
+
+internal object TemporaryDirectory {
+
+    fun use(block: (tmpDir: Path) -> Unit) {
```
Comment (dmitry-zaitsev) [inline maestro-client/src/main/java/maestro/utils/TemporaryDirectory.kt:8]: Nice!

**Review by dmitry-zaitsev — APPROVED** [review]

File: maestro-client/src/main/java/maestro/android/AndroidBuildToolsDirectory.kt (hunk 1 above)
Comment (Leland-Takamine) [reply maestro-client/src/main/java/maestro/android/AndroidBuildToolsDirectory.kt:9]: Good point. I'll add some logic that detects whether the app is already debuggable and if so, `ApkDebuggable.enable` can just copy the src apk to dst location.

Comment (dmitry-zaitsev) [reply maestro-client/src/main/java/maestro/android/AndroidBuildToolsDirectory.kt:9]: Awesome! 👍

**Pull request merged**